cloud/amazon/ec2_ami.py

```diff
@@ -393,7 +393,8 @@ def create_image(module, ec2):
         try:
             img = ec2.get_image(image_id)
 
-            if img.state == 'available':
+            # We must wait for the ami to exist in AWS api responses before we can skip the wait, and add tags when specifying wait: no 
+            if img.state == 'available' or (img and not wait):
```

Comment:
I should mention that if I remove that check that begins on line 406, this works well for me. Thanks!